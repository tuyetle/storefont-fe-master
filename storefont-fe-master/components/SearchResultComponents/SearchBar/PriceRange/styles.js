import { css } from 'glamor';
import variables from '~/styles/variables';

export default {
  priceRange: css({
    width: '34em',
    maxWidth: '598px',
    top: '0em',
    right: '-28em',
    padding: '0 0 1.5em',
    backgroundColor: 'white',

    '@media (min-width: 768px)': {
      position: 'absolute',
      width: '34em',
      maxWidth: '598px',
      top: '0em',
      right: '-29em',
      padding: '1.5em 0 0.5em 1.2em',
      backgroundColor: 'white',
      boxShadow: variables.$boxShadowPrice,
      border: `1px solid ${variables.$borderFilter}`,
    },

    '&div': {
      paddingRight: '0',
      paddingLeft: '0',
    },
    '& .input-group': {
      width: '190px',
      height: '30px',
      border: `0.5px solid ${variables.$borderPrice}`,
      '& .form-control': {
        background: 'none',
        fontSize: '0.875em',
        border: '0',
        fontStyle: 'italic',
        '&:focus': {
          outline: 'none',
          boxShadow: 'none',
        },
      },
      '& .input-group-addon': {
        textTransform: 'uppercase',
        background: 'none',
        border: '0',
        fontSize: '0.8em',
        color: variables.$bgLineMenu,
      },
    },
  }),
  paddingLeft: css({
    verticalAlign: 'middle',
    paddingLeft: '1em',
    marginLeft: '0',
    marginTop: '1em',
    '@media (min-width: 768px)': {
      marginLeft: '-2em',
      marginTop: '0',
    },
  }),
  priceLabel: css({
    float: 'left',
    marginRight: '0.8em',
    fontSize: '0.875em',
    color: variables.$colorWelcome,
    marginTop: '-2.5px',
    width: '2.5em',
    textAlign: 'left',
    '@media (min-width: 768px)': {
      width: 'auto',
      textAlign: 'auto',
    },
  }),
  error: css({
    textAlign: 'left',
    clear: 'both',
    lineHeight: '1.5em',
    color: `${variables.$errorColor}`,
    maxWidth: '21.563em',
  }),
};
