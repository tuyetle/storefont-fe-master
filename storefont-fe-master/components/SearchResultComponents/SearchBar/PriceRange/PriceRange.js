import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Field } from 'react-form';
import { PriceFormField } from '~/components/common/Forms/PriceFormField/PriceFormField';
import styles from './styles';

class PriceRange extends Component {
  onChange = (fieldName, newValue) => {
    const { field, fieldApi, t } = this.props;
    const { values, touched } = fieldApi.getFormState();
    if (!touched || !touched[field]) return;

    if (touched[field].toPrice && fieldName && fieldName[1] === 'fromPrice') {
      const { toPrice } = values[field];
      if (newValue > toPrice) {
        const fromErrorMessage = t('searchResult:fromPriceShouldLessThanToPrice');
        fieldApi.setError(fieldName, fromErrorMessage);
      } else {
        fieldApi.setError(fieldName, null);
        fieldApi.setError([field, 'toPrice'], null);
      }
    } else if (touched[field].fromPrice && fieldName && fieldName[1] === 'toPrice') {
      const { fromPrice } = values[field];
      if (newValue < fromPrice) {
        const toErrorMessage = t('searchResult:toPriceShouldGreaterThanFromPrice');
        fieldApi.setError(fieldName, toErrorMessage);
      } else {
        fieldApi.setError(fieldName, null);
        fieldApi.setError([field, 'fromPrice'], null);
      }
    }
  }

  render() {
    const {
      t, fromPriceLabel, fromPriceDefaultValue,
      toPriceLabel, toPriceDefaultValue,
      billionLabelForUnit, millionLabelForUnit,
      field, fieldApi,
    } = this.props;

    const { errors } = fieldApi.getFormState();
    let priceRangeError = {};
    if (errors && errors[field]) {
      priceRangeError = errors[field];
    }

    return (
      <section className={`${styles.priceRange}`}>
        <div className="row">
          <div className={classnames('col-md-6')}>
            <label className={`${styles.priceLabel}`}>{t('common:from')}</label>
            <Field
              field="fromPrice"
              component={PriceFormField}
              placeholder={fromPriceLabel}
              billionLabelForUnit={billionLabelForUnit}
              millionLabelForUnit={millionLabelForUnit}
              defaultValue={fromPriceDefaultValue}
              onChange={this.onChange}
            />
            {
              priceRangeError.fromPrice && (
                <div className={`${styles.error}`}>
                  {priceRangeError.fromPrice}
                </div>
              )
            }
          </div>

          <div className={classnames(`col-md-6 ${styles.paddingLeft}`)}>
            <label className={`${styles.priceLabel}`}>{t('common:to')}</label>
            <Field
              field="toPrice"
              component={PriceFormField}
              placeholder={toPriceLabel}
              billionLabelForUnit={billionLabelForUnit}
              millionLabelForUnit={millionLabelForUnit}
              defaultValue={toPriceDefaultValue}
              onChange={this.onChange}
            />
            {
              priceRangeError.toPrice && (
                <div className={`${styles.error}`}>
                  {priceRangeError.toPrice}
                </div>
              )
            }
          </div>
        </div>
      </section>
    );
  }
}

PriceRange.propTypes = {
  field: PropTypes.string.isRequired,
  fieldApi: PropTypes.object.isRequired,
  billionLabelForUnit: PropTypes.string.isRequired,
  millionLabelForUnit: PropTypes.string.isRequired,
  fromPriceLabel: PropTypes.string,
  toPriceLabel: PropTypes.string,
  fromPriceDefaultValue: PropTypes.number,
  toPriceDefaultValue: PropTypes.number,
  t: PropTypes.func.isRequired,
};

PriceRange.defaultProps = {
  fromPriceLabel: null,
  toPriceLabel: null,
  fromPriceDefaultValue: 0,
  toPriceDefaultValue: 0,
};

export default PriceRange;
