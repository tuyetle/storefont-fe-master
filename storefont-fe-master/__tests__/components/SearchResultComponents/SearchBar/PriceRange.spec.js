import React from 'react';
import renderer from 'react-test-renderer';
import PriceRange from '~/components/SearchResultComponents/SearchBar/PriceRange/PriceRange';

describe('(Component) PriceRange', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      field: 'sample',
      fieldApi: {
        getFormState() {
          return {};
        },
        setError: () => { },
      },
      billionLabelForUnit: 'TY',
      millionLabelForUnit: 'TR',
      t: () => { },
    };

    component = <PriceRange {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

