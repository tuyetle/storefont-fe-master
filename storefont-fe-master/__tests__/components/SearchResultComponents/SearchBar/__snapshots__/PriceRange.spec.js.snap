// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) PriceRange should render conrectly 1`] = `
<section
  className="css-1cjt15t"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <label
        className="css-legfcw"
      />
      Field
    </div>
    <div
      className="col-md-6 css-17xsdqc"
    >
      <label
        className="css-legfcw"
      />
      Field
    </div>
  </div>
</section>
`;
